docs/ValidationRules/RULEID.RULEFRIENDLYNAME.cs

```diff
@@ -0,0 +1,142 @@
+ *      Where possible, prefer using Imperative language, like `UseAbsoluteUri`
+ *      instead of Indicative language, like `UriIsNotAbsolute`.
+ *      Examples:
+ *          DoNotUseFriendlyNameAsRuleId
+ *          ReferToFinalSchema
+ *
+ * - Rename this file as <RULEID>.<RULEFRIENDLYNAME>.cs.
+ * 
+ * - Remove All INSTRUCTIONS after the changes have been made.
+ */
+namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
+{
+    public class RULEFRIENDLYNAME : SarifValidationSkimmerBase
+    {
+        private readonly MultiformatMessageString _fullDescription = new MultiformatMessageString
+        {
+            /*
+             * INSTRUCTIONS:
+             * Add a new key-value pair in ~\Sarif.Multitool\Rules\RuleResources.resx.
+             * 
+             * Key: 
+             *      RULEID_RULEFRIENDLYNAME
+             *      
+             * Value: 
+             *      Provide a brief description with atleast two sentences, both ending in a period.
+             *      The first sentence should be a short description of the rule.
+             *      The second sentence must describe the utility/usage of the rule.
+             *      
+             * Example:
+             *      The $schema property should be present, and must refer to the final version 
+             *      of the SARIF 2.1.0 schema. This enables IDEs to provide Intellisense for SARIF log files.
+             *      
+             * Notes:
+             *      The first sentence will be used as a `ShortDescription` for this rule.
+             *      All sentences together will be used as a `LongDescription` for this rule.
+             */
+            Text = RuleResources.RULEID_RULEFRIENDLYNAME_Description
+        };
+
+        public override MultiformatMessageString FullDescription => _fullDescription;
+
+        /*
+         * INSTRUCTIONS:
+         * Decide the appropriate FailureLevel for this rule as appropriate.
+         * The following heuristics can be used to arrive at a decision:
+         * 
+         * Error:
+         *      In general, an `Error` should be reserved for rules which address
+         *      a SARIF spec violation that cannot be expressed by the Schema.
+         *      Example:
+         *      SARIF1019.RuleIdMustBePresentAndConsistent
+         *      Per spec, at least one of result.ruleId and result.rule.id must be present, 
+         *      and if both are present, they must be the same.
+         *      
+         * Warning:
+         *      In general, a `Warning` is a good practice which should be followed, but it
+         *      does not violate the SARIF spec.
+         *      Example:
+         *      SARIF1021.DoNotUseFriendlyNameAsRuleId
+         */
+        public override FailureLevel DefaultLevel => FailureLevel.Warning;
+
+
+        /*
+         * INSTRUCTIONS:
+         * Add a new property in ~\src\Sarif.Multitool\Rules\RuleId.cs with 
```

Comment:
>~ [](start = 33, length = 1)

If you're going to use `~` to mean "repo root", then use it consistently. On 14, 37, and 102, you use it without the `src` component.

Reply:
ooh nice! thnk u

---
In reply to: [435562733](https://github.com/microsoft/sarif-sdk/pull/1902#discussion_r435562733) [](ancestors = 435562733)

Reply:
:-) Frankly I'd've preferred it the other way around, since `~` ("home directory") is IMO more plausible for repo root than for source root. But really, you don't need it: `src\Sarif.Mutitool\Rules\RuleId.cs` is fine.

---
In reply to: [435569225](https://github.com/microsoft/sarif-sdk/pull/1902#discussion_r435569225) [](ancestors = 435569225,435562733)